src/gs-screenshot-image.c

```diff
@@ -196,6 +209,59 @@ gs_screenshot_image_show_blurred (GsScreenshotImage *ssimg,
 		gs_image_set_from_pixbuf_with_scale (GTK_IMAGE (ssimg->image2),
 						     pb, (gint) ssimg->scale);
 	}
+
+	ssimg->status = GS_SCREENSHOT_IMAGE_STATUS_BLURRED;
+}
+
+static void
+gs_screenshot_image_save_counterpart (GsScreenshotImage *ssimg, AsImage *im)
+{
+	guint width = ssimg->width;
+	guint height = ssimg->height;
+	gboolean ret;
+	g_autoptr(GError) error = NULL;
+	const char *url = as_image_get_url (im);
+	g_autofree char *filename = NULL;
+	g_autofree char *size_dir = NULL;
+	g_autofree char *cache_kind = NULL;
+	g_autofree char *basename = gs_screenshot_get_cachefn_for_url (url);
+
+	/* if the image is a thumbnail we save the normal size and
+	 * vice-versa */
+	if (width == AS_IMAGE_THUMBNAIL_WIDTH &&
+	    height == AS_IMAGE_THUMBNAIL_HEIGHT) {
+		width = AS_IMAGE_NORMAL_WIDTH;
+		height = AS_IMAGE_NORMAL_HEIGHT;
+	} else {
+		width = AS_IMAGE_THUMBNAIL_WIDTH;
+		height = AS_IMAGE_THUMBNAIL_HEIGHT;
+	}
+
+	size_dir = g_strdup_printf ("%ux%u", width * ssimg->scale,
+				    height * ssimg->scale);
```

Comment:
You could save scaled width and height in separate variables, since they're used more than once